glean_parser/javascript_server.py

```diff
@@ -28,11 +38,14 @@
 
 # Adding a metric here will require updating the `generate_js_metric_type` function
 # and might require changes to the template.
-SUPPORTED_METRIC_TYPES = ["string"]
+SUPPORTED_METRIC_TYPES = ["string", "event"]
 
 
-def event_class_name(pingName: str) -> str:
-    return util.Camelize(pingName) + "ServerEvent"
+def event_class_name(pingName: str, event_metric_exists: bool) -> str:
+    # For compatibility with FxA codebase we don't want to add "Logger" suffix
+    # when custom pings without event metrics are used.
+    suffix = "Logger" if event_metric_exists else ""
+    return util.Camelize(pingName) + "ServerEvent" + suffix
```

Comment:
Uff, seeing both camelCase and snake_case mixed here. Should we clean this up while we're here?

```suggestion
def event_class_name(ping_name: str, event_metric_exists: bool) -> str:
    # For compatibility with FxA codebase we don't want to add "Logger" suffix
    # when custom pings without event metrics are used.
    suffix = "Logger" if event_metric_exists else ""
    return util.Camelize(ping_name) + "ServerEvent" + suffix
```